core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerChannelManager.scala

```diff
@@ -285,15 +280,16 @@ class TransactionMarkerChannelManager(config: KafkaConfig,
     }
   }
 
-  private def maybeWriteTxnCompletion(transactionalId: String): Unit = {
-    Option(transactionsWithPendingMarkers.get(transactionalId)).foreach { pendingCommitTxn =>
```

Comment:
Multiple threads may see the transaction still as pending and attempt completion.